### PR TITLE
Change all links to https

### PR DESCRIPTION
**Pull request by Baiqiang**
All links should have the same schema and hostname.

### CODE REVIEW

File: WcaOnRails/app/controllers/competitions_controller.rb [hunk 1]
```diff
@@ -91,20 +91,20 @@ def post_results
     top333 = comp.results.where(eventId: '333', roundId: ['f', 'c']).order(:pos).limit(3)
     if top333.empty? # If there was no 3x3x3 event.
       title = "Results of #{comp.name} posted"
-      body = "Results of the [#{comp.name}](http://www.worldcubeassociation.org/results/c.php?i=#{comp.id}) are now available.\n\n"
+      body = "Results of the [#{comp.name}](https://www.worldcubeassociation.org/results/c.php?i=#{comp.id}) are now available.\n\n"
```
Comment (jfly) [inline WcaOnRails/app/controllers/competitions_controller.rb]: This reminds me that all our old posts will still have http links. How OCD are we? Do we want to write a database migration to update all old posts?

**Comment by jfly** [comment]
Thanks @Baiqiang! This looks great, one question inline.

File: WcaOnRails/app/controllers/competitions_controller.rb (hunk 1 above)
Comment (Baiqiang) [inline WcaOnRails/app/controllers/competitions_controller.rb]: I think we should do the migration to keep all links in one format.

Comment (Baiqiang) [inline WcaOnRails/app/controllers/competitions_controller.rb]: I think we should do the migration to keep all links in one format.

**Pull request merged**